### PR TITLE
Refactored select renderer code to be a little easier to comprehend

### PR DESCRIPTION
**Pull request by sandersky**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

* Refactored select renderer code to be a little easier to comprehend.

### CODE REVIEW

File: addon/components/inputs/select.js [hunk 1]
```diff
@@ -116,81 +134,95 @@ export default AbstractInput.extend({
...
-    const cellConfig = this.get('cellConfig')
-    const modelDef = this.get('bunsenModel')
-    const options = _.get(cellConfig, 'renderer.options')
+  /**
+   * Determine if query has references to properties that have yet to be set.
+   * If the query contains no references then there is nothing to wait on.
+   * @param {Object} formValue - form value
+   * @returns {Boolean} whether or not query is waiting on missing references
+   */
+  iswaitingOnReferences (formValue) {
```
Comment (sophypal) [inline addon/components/inputs/select.js]: That's a much better name than `isQueryDisabled` but I'd prefer `isWaitingOnReferences`.

File: addon/components/inputs/select.js [hunk 2]
```diff
@@ -116,81 +134,95 @@ export default AbstractInput.extend({
...
-  },
 
-  hasQueryParams (query) {
-    if (!query || _.isEmpty(query)) {
-      return false
+    // If the query has changed or we have yet to initialize the items lets go
+    // get our items
+    if (
+      this.hasQueryChanged(oldValue, newValue, options.query) ||
+      this.needsInitialItems()
```
Comment (sophypal) [inline addon/components/inputs/select.js]: I fear that evaluating `this.needsInitialItems()` after `isWaitingOnReferences` will prevent the initial fetch for selects that don't have references but have query defined.

**Comment by sandersky** [comment]
:+1:

File: addon/components/inputs/select.js (hunk 1 above)
Comment (sandersky) [inline addon/components/inputs/select.js]: Ah dammit. Fixing in next PR.

File: addon/components/inputs/select.js (hunk 2 above)
Comment (sandersky) [inline addon/components/inputs/select.js]: I'll revert it.

**Pull request merged**